packages/shared/test/stress-test.ts

```diff
@@ -0,0 +1,176 @@
+
+  return response.json();
+}
+
+function updateFixture(recordingId: string, fixture: Fixture) {
+  if (!fs.existsSync(getFixturePath())) {
+    fs.mkdirSync(getFixturePath());
+  }
+  fs.writeFileSync(getFixturePath(recordingId), JSON.stringify(fixture, null, 2));
+}
+
+// The fixture is a simple data structure that contains
+// the recording metadata, annotations returned from the protocol, and the result
+async function fetchFixture(recordingId: string, client: ReplayClient): Promise<Fixture> {
+  const sessionId = await client.initialize(recordingId, process.env.STRESS_TEST_API_KEY!);
+  debug("sessionId", sessionId);
+
+  if (fs.existsSync(getFixturePath(recordingId))) {
+    return JSON.parse(fs.readFileSync(getFixturePath(recordingId), "utf8"));
+  }
+
+  const annotations: null | any[] = null;
+  const recording = await graphqlQuery(
+    `
+      query GetRecording($recordingId: UUID!) {
+        recording(uuid: $recordingId) {
+          id
+          metadata
+        }
+      }
+      `,
+    { recordingId }
+  );
+
+  return { metadata: recording.data.recording.metadata, annotations, result: null };
+}
+
+// The proxy client is responsible for proxying calls to findAnnotations
+// such that we can either read from the fixture or fetch from the server
+function createProxyClient(client: ReplayClient, fixture: Fixture) {
+  return new Proxy(client, {
+    get: function (target, prop, receiver) {
+      if (prop == "findAnnotations") {
+        return async (type: string, callback: (annotation: any) => void) => {
+          if (type !== "replay-cypress") {
+            return target.findAnnotations(type, callback);
+          }
+
+          // If the fixture does not have annotations lets fetch them
+          // and save them to the fixture
+          if (!fixture.annotations) {
+            debug("fetching annotations");
+            fixture.annotations = [];
+            return target.findAnnotations(type, (annotation: any) => {
+              fixture.annotations!.push(annotation);
+              callback(annotation);
+            });
+          }
+
+          // If the fixture has annotations lets just call the callback with them
+          debug("reading annotations from cache");
+          for (const annotation of fixture.annotations) {
+            callback(annotation);
+          }
+        };
+      }
+
+      return Reflect.get(target, prop, receiver);
+    },
+  });
+}
+
+async function testRecording(recordingId: string) {
+  const client = new ReplayClient("wss://dispatch.replay.io");
+
+  try {
+    debug("testRecording", recordingId);
+
+    const fixture = await fetchFixture(recordingId, client);
+    const result = await processGroupedTestCases(
+      fixture.metadata.test,
+      createProxyClient(client, fixture)
+    );
+
+    updateFixture(recordingId, { ...fixture, result });
+    await client.releaseSession();
+
+    if (fixture.result) {
+      return isEqual(fixture.result, result);
+    }
+
+    return true;
```

Comment:
it's nice that the test actually fits in a couple lines of code. The rest is just boilerplate